packages/lit-auth-client/src/lib/providers/DiscordProvider.ts

```diff
@@ -0,0 +1,87 @@
+import { AuthMethod, OAuthProviderOptions } from '@lit-protocol/types';
+import { AuthMethodType } from '@lit-protocol/constants';
+import { BaseProvider } from './BaseProvider';
+import {
+  prepareLoginUrl,
+  parseLoginParams,
+  getStateParam,
+  decode,
+} from '../utils';
+
+export default class DiscordProvider extends BaseProvider {
+  /**
+   * The redirect URI that Lit's login server should send the user back to
+   */
+  public redirectUri: string;
+
+  constructor(options: OAuthProviderOptions) {
+    super(options);
+    this.redirectUri = options.redirectUri;
+  }
+
+  /**
+   * Redirect user to the Lit's Discord login page
+   *
+   * @returns {DiscordProvider} - Redirects user to Lit login page
+   */
+  public signInWithLit(): void {
+    // Get login url
+    const loginUrl = prepareLoginUrl('discord', this.redirectUri);
+    // Redirect to login url
+    window.location.assign(loginUrl);
+  }
+
+  /**
+   * Validate the URL parameters returned from Lit's login server and return the OAuth token
+   *
+   * @returns {AuthMethod} - Auth method object containing the OAuth token
+   */
+  public authenticateWithLit(): AuthMethod {
```

Comment:
perhaps we uniform this to `authenticate` and add to `BaseProvider`

Reply:
@joshLong145 - updated this to implement the `BaseProvider`'s `authenticate`